src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/AvailabilityService.kt

```diff
@@ -89,47 +96,45 @@ class AvailabilityService(
     return availabilityFinderService.getOptions(
       request,
       bvlsAppointmentSlotsToInclude.plus(externalAppointmentSlotsToInclude),
-      requestedLocations,
+      requestedLocations.values.toList(),
     )
   }
 
-  /**
-   * Assumptions:
-   *  - we consider ONLY video link bookings (not other appointment types from NOMIS or A&A in these locations)
-   *  - we consider ONLY bookings present in the BVLS service, not in A&A or NOMIS (i.e. all video link bookings)
-   *  - current booking journeys for BVLS allows only one person at one prison per booking.
-   *  - we consider ONLY the prison locations in the request for alternative times (we do not consider others - yet)
-   *  - the decision about whether a user is allowed to book into a locations is still the users own.
-   *  - we assume all appointments for this booking are on the same day and at the same prison
-   *
-   * To consider later:
-   *  - Get other VCC-enabled rooms at these prison(s) and offer them?
-   *  - Incorporate prison room decoration logic here, to find other rooms?
-   */
-  fun checkAvailabilityVlbOnly(request: AvailabilityRequest): AvailabilityResponse {
-    // Gather the distinct list of locations from the request
-    val locationKeys = setOfNotNull(
-      request.preAppointment?.prisonLocKey,
-      request.postAppointment?.prisonLocKey,
-      request.mainAppointment!!.prisonLocKey,
-    )
+  private fun AvailabilityRequest.isForAnExistingBooking() = vlbIdToExclude != null
 
```

Comment:
^ Is this always supplied when its for an existing booking?

Reply:
During amend yes.  Of course there is no guarantee but this is what we have to work with for now.